### ISSUE TITLE
Array Merge error (and Ambiguous class detection)

### ISSUE THREAD

**Issue by PeterJC91** [intrip/laravel-authentication-acl#103]
Hi,

First of all: Nice work on the package!

Now I have installed the L5.1 version and want to use it, but I get an error after I add the serviceprovider and clean up my installation (dump-autoload, clear-compiled, config:cache, route:cache, optimize --force).

I get an error about the array merge in vendor\laravel\framework\src\illuminte\support\serviceprovider (in function mergeConfigFrom). I see that the $config var is Null all of a sudden. Initializing it with an empty error fixes the problem, but then when I want to use the authenticator I'll get an invalid Hasher.

"ErrorException in ServiceProvider.php line 68: array_merge(): Argument #2 is not an array"

Any ideas what could be causing this (and how to resolve this)?

Ps: I also get a lot of Ambiguous class resolutions, could that have anything to do with it?

1) Warning: Ambiguous class resolution, "MigrationCartalystSentryInstallUsers" was
found in both "vendor/jacopo/laravel-authentication-acl/database/migrations/2012_12_06_225921_migration_cartalyst_sentry_install_users.php" and "vendor\jacopo\authentication-sentry\src\migrations\2012_12_06_225921_migration_cartalyst_sentry_install_users.php", the first will be used.

2) Warning: Ambiguous class resolution, "MigrationCartalystSentryInstallGroups" was
 found in both "vendor/jacopo/laravel-authentication-acl/database/migrations/2012_12_06_225929_migration_cartalyst_sentry_install_groups.php" and "vendor\jacopo\authentication-sentry\src\migrations\2012_12_06_225929_migration_cartalyst_sentry_install_groups.php", the first will be used.

3) Warning: Ambiguous class resolution, "MigrationCartalystSentryInstallUsersGroups
Pivot" was found in both "vendor/jacopo/laravel-authentication-acl/database/migrations/2012_12_06_225945_migration_cartalyst_sentry_install_users_groups_pivot.php" and "vendor\jacopo\authentication-sentry\src\migrations\2012_12_06_225945_migration_cartalyst_sentry_install_users_groups_pivot.php", the first will be used.

4) Warning: Ambiguous class resolution, "MigrationCartalystSentryInstallThrottle" w
as found in both "vendor/jacopo/laravel-authentication-acl/database/migrations/2012_12_06_225988_migration_cartalyst_sentry_install_throttle.php" and "vendor\jacopo\authentication-sentry\src\migrations\2012_12_06_225988_migration_cartalyst_sentry_install_throttle.php", the first will be used.

Update: I am trying to see what happens if I build a new Laravel app and ad everything back in there. Will come back with the results once it is done.

**Comment by PeterJC91** [comment]
Update:
Oke I have completely rebuild my application several times. I can install everything and then when I want to use the authenticator I get the error that the class authenticator is not found. So I stop everything and start 'optimizing':

dump-autoload works
clear-compiler works
config:cache works
route:cache -> here the error comes:

array_merge(): Argument #2 is not an array.

And then I can do nothing with the app anymore. Even uncommenting the ServiceProvider in the app.php and commenting the authenticator line and optimizing won't fix the problem.

Any ideas?

**Comment by intrip** [comment]
Could you write your service_providers.php line that includes the AuthenticationServiceProvider?

**Comment by PeterJC91** [comment]
I have rebuild my application like 6 times now and eventually it works now. I have no idea why it didn't work the first couple of time, but no need to put more effort in it. So let's just close this issue. Apologies for the inconvience and thanks for your time.

**Comment by ghost** [comment]
Regarding the ambiguous classes, @skeser noted [on 6 Apr 2016](https://github.com/intrip/laravel-authentication-acl/issues/108#issuecomment-206537573) and @ashish64 noted [on 6 Nov 2016](https://github.com/intrip/laravel-authentication-acl/issues/151#issuecomment-258671664) that a workaround is to delete Laravel's `create_password_resets_table` and `create_users_table` migrations from the `database/migrations` folder.
